File: data-skill/templates/python/process_data.py
#!/usr/bin/env python3
# Data Processing Template
# Purpose: Filter, recode, transform, and aggregate data

import pandas as pd

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file = "data.csv"
# ============================================================================

# Load data
df = pd.read_csv(data_file)

# ============================================================================
# DATA PROCESSING PIPELINE
# ============================================================================
plot_data = df.copy()

# 1. Filter observations
# plot_data = plot_data[plot_data['variable_name'].notna()]

# 2. Mutate / Recode variables
plot_data = plot_data.assign(
    # Example: Chinese to English labels
    # new_var=plot_data['variable_name'].map({
    #     '中文值1': 'English Label 1',
    #     '中文值2': 'English Label 2'
    # }),

    # Example: Create binary variable
    # binary_var=plot_data['variable_name'].apply(lambda x: 1 if x == '是' else 0),

    # Example: Calculate derived variable
    # percentage=lambda x: x['n'] / x['n'].sum() * 100
)

# 3. Group and aggregate
# plot_data = plot_data.groupby('grouping_var').agg({
#     'numeric_var': ['count', 'mean', 'std']
# }).reset_index()

# 4. Arrange
# plot_data = plot_data.sort_values('count', ascending=False)

# ============================================================================
# VERIFY OUTPUT
# ============================================================================
print("\n" + "=" * 40)
print("PROCESSED DATA")
print("=" * 40 + "\n")

print(plot_data)

print(f"\nDimensions: {plot_data.shape[0]} rows x {plot_data.shape[1]} cols")
print(f"Column names: {list(plot_data.columns)}")

# Export for plotting
# plot_data.to_csv("plot_data.csv", index=False)

print("\n" + "=" * 40)

File: data-skill/templates/r/process_data.R
#!/usr/bin/env Rscript
# Data Processing Template
# Purpose: Filter, recode, transform, and aggregate data

library(tidyverse)

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file <- "data.csv"
# ============================================================================

# Load data
df <- read_csv(data_file)

# ============================================================================
# DATA PROCESSING PIPELINE
# ============================================================================
plot_data <- df %>%
  # 1. Filter observations
  # filter(!is.na(variable_name)) %>%

  # 2. Mutate / Recode variables
  mutate(
    # Example: Chinese to English labels
    # new_var = case_when(
    #   variable_name == "中文值1" ~ "English Label 1",
    #   variable_name == "中文值2" ~ "English Label 2",
    #   TRUE ~ as.character(variable_name)
    # ),

    # Example: Create binary variable
    # binary_var = ifelse(variable_name == "是", 1, 0),

    # Example: Calculate derived variable
    # percentage = n / sum(n) * 100
  ) %>%

  # 3. Group and summarize
  # group_by(grouping_var) %>%
  # summarise(
  #   count = n(),
  #   mean_value = mean(numeric_var, na.rm = TRUE),
  #   .groups = "drop"
  # ) %>%

  # 4. Arrange
  # arrange(desc(count))

# ============================================================================
# VERIFY OUTPUT
# ============================================================================
cat("\n========================================\n")
cat("PROCESSED DATA\n")
cat("========================================\n\n")

print(plot_data)

cat("\nDimensions: ", nrow(plot_data), " rows x ", ncol(plot_data), " cols\n")
cat("Column names: ", names(plot_data), "\n")

# Export for plotting
# write_csv(plot_data, "plot_data.csv")

cat("\n========================================\n")
